OpenRA.Mods.Common/Activities/DeliverResources.cs

```diff
@@ -48,6 +53,11 @@ public override bool Tick(Actor self)
 			if (IsCanceling)
 				return true;
 
+			if (wasMobileMoved && mobile != null && mobile.MoveResult == MoveResult.MovementStuck)
+				return true;
+			else
+				wasMobileMoved = false;
```

Comment:
For all activities, we need to decide what policy to apply. Previously, activities all seemed to employ a "retry as fast as possible" technique, and so would cause performance to plummet.

I've noted some potiential policies for Enter and Attack. Both could be allowed to retry, but only after a delay to avoid spamming pathfinding too much.

Some activities might want the policy this PR has applied, which is to give up as soon as the move attempt has completed. That might be quite reasonable if retrying automatically isn't likely to work.

Some activities might be able to do something clever if the destination was reached versus if it was blocked. e.g. maybe `DeliverResources` could retry at another nearby refinery instead if the current one is blocked? We don't need that in this PR, I'm just making the point that the activity could use that information to do different things. Maybe the attack activities give up but ping a notification to the player that they got stuck? Things like that.